adodb.inc.php

```diff
@@ -3925,6 +3918,36 @@ function __toString() {
 		return _adodb_export($this,',',',',false,true);
 	}
 
+	function _initrs() {
+		$this->_numOfRows =  sizeof($this->_array);
+		if ($this->_skiprow1) {
+			$this->_numOfRows -= 1;
+		}
+
+		$this->_numOfFields = (isset($this->_fieldobjects))
+			? sizeof($this->_fieldobjects)
+			: sizeof($this->_types);
+	}
+
+	/**
+	 * @return bool
+	 */
+	function _fetch() {
+		$pos = $this->_currentRow;
+
+		if ($this->_numOfRows <= $pos) {
+			if (!$this->compat) {
+				$this->fields = false;
+			}
+			return false;
+		}
+		if ($this->_skiprow1) {
+			$pos += 1;
+		}
+		$this->fields = $this->_array[$pos];
+		return true;
+	}
+
```

Comment:
This introduces a syntax error 
> PHP Fatal error:  Cannot redeclare ADORecordSet::_initRS() in adodb.inc.php on line 3974

Not sure why you are redefining _initRS() and _fetch() methods... And where is this code coming from ?